src/components/dashboard/Orders.tsx

```diff
@@ -1,93 +1,283 @@
-import * as React from "react";
-import Link from "@mui/material/Link";
-import Table from "@mui/material/Table";
-import TableBody from "@mui/material/TableBody";
-import TableCell from "@mui/material/TableCell";
-import TableHead from "@mui/material/TableHead";
-import TableRow from "@mui/material/TableRow";
-import Title from "./Title";
-
-// Generate Order Data
-function createData(
-    id: number,
-    date: string,
-    name: string,
-    shipTo: string,
-    paymentMethod: string,
-    amount: number,
-) {
-    return { id, date, name, shipTo, paymentMethod, amount };
-}
+import React, { useState, useEffect, ChangeEvent } from "react";
+import Box from "@mui/material/Box";
+import Divider from "@mui/material/Divider";
+import Button from "@mui/material/Button";
+import Stack from "@mui/material/Stack";
+import TextField from "@mui/material/TextField";
+import SearchOutlinedIcon from "@mui/icons-material/SearchOutlined";
+import InputAdornment from "@mui/material/InputAdornment";
+import FormControl from "@mui/material/FormControl";
+import Paper from "@mui/material/Paper";
+import { Typography } from "@mui/material";
+import * as XLSX from "xlsx";
+import {
+    DataGrid, GridColDef, gridPageCountSelector,
+    gridPageSelector,
+    useGridApiContext,
+    useGridSelector,
+} from "@mui/x-data-grid";
+import Pagination from "@mui/material/Pagination";
+import PaginationItem from "@mui/material/PaginationItem";
+
+
+
+
+
+//! Helper function for custom Pagination 
+function CustomPagination() {
+    const apiRef = useGridApiContext();
```

Comment:
Why are you building custom pagination? MUI Data Grid has built in pagination

Reply:
In order to make pagination style like what is required, I've built custom pagination.